packages/react-reconciler/src/SchedulingProfiling.js

```diff
@@ -0,0 +1,189 @@
+  if (enableSchedulingProfiling) {
+    if (supportsUserTiming) {
+      performance.mark(`--commit-stop`);
+    }
+  }
+}
+
+const PossiblyWeakMap = typeof WeakMap === 'function' ? WeakMap : Map;
+// $FlowFixMe: Flow cannot handle polymorphic WeakMaps
+const wakeableIDs: WeakMap<Wakeable, number> = new PossiblyWeakMap();
+let wakeableID: number = 0;
+function getWakeableID(wakeable: Wakeable): number {
+  if (!wakeableIDs.has(wakeable)) {
+    wakeableIDs.set(wakeable, wakeableID++);
+  }
+  return ((wakeableIDs.get(wakeable): any): number);
+}
+
+export function markComponentSuspended(
+  componentName: string,
+  fiber: Fiber,
+  wakeable: Wakeable,
+): void {
+  if (enableSchedulingProfiling) {
+    if (supportsUserTiming) {
+      const id = getWakeableID(wakeable);
+      const componentStack = getStackByFiberInDevAndProd(fiber) || '';
+      // TODO (brian) Generate and store temporary ID so DevTools can match up a component stack later.
+      performance.mark(
+        `--suspense-suspend-${componentName}-${id}-${componentStack}`,
+      );
+      wakeable.then(
+        () =>
+          performance.mark(
+            `--suspense-resolved-${componentName}-${id}-${componentStack}`,
+          ),
+        () =>
+          performance.mark(
+            `--suspense-rejected-${componentName}-${id}-${componentStack}`,
+          ),
+      );
+    }
+  }
+}
+
+export function markLayoutEffectsStarted(lanes: Lanes): void {
+  if (enableSchedulingProfiling) {
+    if (supportsUserTiming) {
+      performance.mark(`--layout-effects-start-${formatLanes(lanes)}`);
+    }
+  }
+}
+
+export function markLayoutEffectsStopped(): void {
+  if (enableSchedulingProfiling) {
+    if (supportsUserTiming) {
+      performance.mark('--layout-effects-stop');
+    }
+  }
+}
+
+export function markPassiveEffectsStarted(lanes: Lanes): void {
+  if (enableSchedulingProfiling) {
+    if (supportsUserTiming) {
+      performance.mark(`--passive-effects-start-${formatLanes(lanes)}`);
+    }
+  }
+}
+
+export function markPassiveEffectsStopped(): void {
+  if (enableSchedulingProfiling) {
+    if (supportsUserTiming) {
+      performance.mark('--passive-effects-stop');
+    }
+  }
+}
+
+export function markRenderStarted(lanes: Lanes): void {
+  if (enableSchedulingProfiling) {
+    if (supportsUserTiming) {
+      performance.mark(`--render-start-${formatLanes(lanes)}`);
+    }
+  }
+}
+
+export function markRenderYielded(): void {
+  if (enableSchedulingProfiling) {
+    if (supportsUserTiming) {
+      performance.mark('--render-yield');
+    }
+  }
+}
+
+export function markRenderStopped(): void {
+  if (enableSchedulingProfiling) {
+    if (supportsUserTiming) {
+      performance.mark('--render-stop');
+    }
+  }
+}
+
+export function markRenderScheduled(
+  componentName: string,
+  fiber: Fiber,
+  lane: Lane,
+): void {
+  if (enableSchedulingProfiling) {
+    if (supportsUserTiming) {
+      const componentStack = getStackByFiberInDevAndProd(fiber) || '';
```

Comment:
I don't think we need a component stack here, because the `fiber` in this case is the root most Fiber. Let's just show its name.

Reply:
I've removed both the component stack and the name because of the earlier comment https://github.com/facebook/react/pull/19223#discussion_r448383821. I don't have a clear understanding of what the `current` fiber that's passed to `markRenderScheduled` is so I can't really reason about what's happening, but I'm definitely only seeing an "Unknown" component in all my render scheduled calls. Let me know if I should add the component name back :)

Reply:
Nope, this is fine 👍